### ISSUE TITLE
GLFW does not have the INTERFACE_LINK_LIBRARIES property set

### ISSUE THREAD

**Issue by malaterre** [AcademySoftwareFoundation/openvdb#635]
When building openvdb on Debian buildd machine I can see the following warning:

```
-- Found glfw3: /usr/lib/powerpc-linux-gnu/cmake/glfw3 (found suitable version "3.3.2", minimum required is "3.1") 
CMake Warning at cmake/OpenVDBGLFW3Setup.cmake:138 (message):
  GLFW does not have the INTERFACE_LINK_LIBRARIES property set, so
  hard-coding to expect a dependency on X11.  To use a different library
  dependency, consider upgrading to GLFW 3.2+ where this property is set by
  the CMake find package module for GLFW.
Call Stack (most recent call first):
  openvdb/cmd/CMakeLists.txt:203 (include)
```

I fail to understand what the warning is telling me. The version of the package for glfw3 is 3.3.2 (as indicated above), but the warning seems to imply that I should instead be using something later than 3.2+... which does not make sense to me.

For reference I am using cmake version 3.16.3

Full ref:

* https://buildd.debian.org/status/fetch.php?pkg=openvdb&arch=powerpc&ver=6.2.1-4&stamp=1581930003&raw=0

**Comment by Idclip** [comment]
@danrbailey you able to look at this one?

**Comment by danrbailey** [comment]
Hi @malaterre, are you able to share what this generated CMake file looks like for you:

`/usr/lib/powerpc-linux-gnu/cmake/glfw3/glfw3Targets.cmake`

In particular, I just compiled and installing GLFW 3.3 from source and I get a section that looks like this:

```
set_target_properties(glfw PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "..."
  INTERFACE_LINK_LIBRARIES "..."
)
```

If I install GLFW 3.1, these lines don't exist which is what this logic is intended to handle (they show up for the first time in 3.2).

**Comment by malaterre** [comment]
Here is what I have (Debian sid):

```
$ cat /usr/lib/x86_64-linux-gnu/cmake/glfw3/glfw3Targets.cmake | grep -A 3 set_target_properties 
set_target_properties(glfw PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
)


```

**Comment by danrbailey** [comment]
Intriguing. I think I'll change this logic based on the GLFW version instead. My guess is that if INTERFACE_LINK_LIBRARIES is _not_ required on a particular arch, it doesn't get added at all so this warning is not necessarily correct.

**Comment by danrbailey** [comment]
Think has now been fixed (#656), please reopen if this is still an issue.
